### PR TITLE
CI: install qemu as part of CI scripts on ppc64le

### PR DESCRIPTION
**Pull request by nitkon**
Add scripts to install qemu as part of the CI.

Fixes: #1068

Signed-off-by: Nitesh Konkar niteshkonkar@in.ibm.com

### CODE REVIEW

File: .ci/ppc64le/lib_install_qemu_ppc64le.sh [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+                sudo dnf install -y "$PACKAGED_QEMU"
+	elif [ "$ID" == "centos" ]; then
+		sudo yum install -y "$PACKAGED_QEMU"
+        else
+                die "Unrecognized distro"
+        fi
+}
+
+build_and_install_qemu() {
+	echo "inside new ppc64le build and install"
```
[the remark this answers is not in the document]
Comment (nitkon) [reply .ci/ppc64le/lib_install_qemu_ppc64le.sh]: unintentional, let me remove it.

File: .ci/ppc64le/lib_install_qemu_ppc64le.sh [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+#
+# SPDX-License-Identifier: Apache-2.0
+
+set -e
+
+CURRENT_QEMU_VERSION=$(get_version "assets.hypervisor.qemu.version")
+PACKAGED_QEMU="qemu"
+
+get_packaged_qemu_version() {
+        if [ "$ID" == "ubuntu" ]; then
```
[the remark this answers is not in the document]
Comment (nitkon) [reply .ci/ppc64le/lib_install_qemu_ppc64le.sh]: Yes, we support all these distros on our platform.

File: .ci/ppc64le/lib_install_qemu_ppc64le.sh [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+        PACKAGING_REPO="github.com/kata-containers/packaging"
+        QEMU_CONFIG_SCRIPT="${GOPATH}/src/${PACKAGING_REPO}/scripts/configure-hypervisor.sh"
+
+        go get -d "${QEMU_REPO}" || true
+        go get -d "$PACKAGING_REPO" || true
+
+        pushd "${GOPATH}/src/${QEMU_REPO}"
+        git fetch
+        git checkout "$CURRENT_QEMU_VERSION"
+        [ -d "capstone" ] || git clone https://github.com/qemu/capstone.git capstone
```
Comment (devimc) [inline .ci/ppc64le/lib_install_qemu_ppc64le.sh]: not sure, but I remember that qemu 2.x can't build with capstone/keycodemapdb master

Comment (devimc) [reply .ci/ppc64le/lib_install_qemu_ppc64le.sh]: @chavafg  https://github.com/kata-containers/packaging/blob/master/snap/snapcraft.yaml#L279-L283
 
let's merge it and see what the CI says

File: .ci/ppc64le/lib_install_qemu_ppc64le.sh [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+        echo "Build Qemu"
+        "${QEMU_CONFIG_SCRIPT}" "qemu" | xargs ./configure
+        make -j $(nproc)
+
+        echo "Install Qemu"
+        sudo -E make install
+
+        # Add link from /usr/local/bin to /usr/bin
+        sudo ln -sf $(command -v qemu-system-${QEMU_ARCH}) "/usr/bin/qemu-system-${QEMU_ARCH}"
```
Comment (chavafg) [inline .ci/ppc64le/lib_install_qemu_ppc64le.sh]: @nitkon now that the CI ran with this, I see that we don't need this line as qemu is installed in `/usr/bin` by default
http://jenkins.katacontainers.io/job/kata-containers-proxy-Power8-16.04-PR-initrd/23/console

**Pull request merged**